src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -21,255 +169,1401 @@ const ActorInit En_Nb_InitVars = {
+    EnNb_SetPosRotInPortal(this, globalCtx);
+    EnNb_CheckKidnapCsMode(this, globalCtx);
+}
+
+void EnNb_SetupConfrontation(EnNb* this, GlobalContext* globalCtx) {
+    AnimationHeader* animation = &D_06008BD0;
+    EnNb_SetCurrentAnim(this, animation, 0, 0.0f, 0);
+    this->action = NB_IN_CONFRONTATION;
+    this->actor.shape.unk_14 = 0;
+}
+
+void EnNb_PlayKnuckleDefeatSFX(EnNb* this, GlobalContext* globalCtx) {
+    Vec3f* projectedPos = &this->actor.projectedPos;
+    s32 pad;
+
+    if (globalCtx->csCtx.frames == 548) {
+        func_80078914(projectedPos, NA_SE_VO_NB_CRY_0);
+        func_80078914(projectedPos, NA_SE_EN_FANTOM_HIT_THUNDER);
+    }
+}
+
+void EnNb_PlayKneelingOnGroundSFX(EnNb* this) {
+    s32 pad;
+    SkelAnime* skelAnime = &this->skelAnime;
+
+    if ((skelAnime->mode == 2) && ((func_800A56C8(skelAnime, 18.0f) != 0) || (func_800A56C8(skelAnime, 25.0f) != 0))) {
+        func_80078914(&this->actor.projectedPos, NA_SE_EV_HUMAN_BOUND);
+    }
+}
+
+void EnNb_PlayLookRightSFX(EnNb* this) {
+    s32 pad;
+    SkelAnime* skelAnime = &this->skelAnime;
+
+    if ((skelAnime->mode == 2) && (func_800A56C8(skelAnime, 9.0f) != 0)) {
+        func_80078914(&this->actor.projectedPos, NA_SE_PL_WALK_CONCRETE);
+    }
+}
+
+void EnNb_PlayLookLeftSFX(EnNb* this) {
+    s32 pad;
+    SkelAnime* skelAnime = &this->skelAnime;
+
+    if ((func_800A56C8(skelAnime, 9.0f) != 0) || (func_800A56C8(skelAnime, 13.0f) != 0)) {
+        func_80078914(&this->actor.projectedPos, NA_SE_PL_WALK_CONCRETE);
+    }
+}
+
+void EnNb_InitDemo6KInConfrontation(EnNb* this, GlobalContext* globalCtx) {
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DEMO_6K, this->actor.posRot.pos.x,
+                kREG(21) + 22.0f + this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 0xB);
+}
+
+void func_80AB2688(EnNb* this, GlobalContext* globalCtx) {
+    this->skelAnime.flags |= 1;
+    SkelAnime_LoadAnimationType5(globalCtx, &this->actor, &this->skelAnime, 1.0f);
+}
+
+void func_80AB26C8(EnNb* this) {
+    this->action = NB_IN_CONFRONTATION;
+    this->drawMode = NB_DRAW_NOTHING;
+    this->actor.shape.unk_14 = 0;
+}
+
+void func_80AB26DC(EnNb* this, GlobalContext* globalCtx) {
+    s32 pad;
+    AnimationHeader* animation = &D_06008BD0;
+    f32 frames = SkelAnime_GetFrameCount(&animation->genericHeader);
+
+    EnNb_SetupCsPosRot(this, globalCtx, 1);
+    SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f, frames, 2, 0.0f);
+    this->action = NB_ACTION_14;
+    this->drawMode = NB_DRAW_KNEEL;
+    this->actor.shape.unk_14 = 0xFF;
+}
+
+void EnNb_SetupKneel(EnNb* this) {
+    AnimationHeader* animation = &D_06008BD0;
+    f32 frames = SkelAnime_GetFrameCount(&animation->genericHeader);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f, frames, 2, 0.0f);
+    this->action = NB_KNEEL;
+    this->drawMode = NB_DRAW_KNEEL;
+    this->actor.shape.unk_14 = 0xFF;
+}
+
+void EnNb_CheckIfKneeling(EnNb* this, UNK_TYPE arg1) {
```

Comment:
how did you decide to name this? `EnNb_SetRaisedArmCaptureAnim` looks very close and has a completely different name

Reply:
I looked at the animation in an animation viewer

Reply:
I meant the `CheckIf` part rather than the `Kneeling` part, since `EnNb_SetRaisedArmCaptureAnim` uses a pattern looking like `EnNb_Set[anim_desc]Anim`